### PR TITLE
Add FFT implementation without graphs

### PR DESCRIPTION
**Pull request by lukasm91**
This PR is adding an implementation to call FFT functionality without Graphs, especially for debugging purposes.

This is adding a new option 
GPU_GRAPHS_FFT
similar to the GEMM and also an implementation similar to the one in FFT.

Also, I redo clang-format (with the default settings), maybe we can also just provide a clang-format file?

### CODE REVIEW

File: src/trans/gpu/algor/hicfft.hip.cpp [hunk 1]
```diff
@@ -34,9 +34,38 @@ struct Float {
...
+    if constexpr (Direction == HIPFFT_R2C)
+      fftSafeCall(hipfftExecR2C(handle, data_real_l, data_complex_l));
+    else if constexpr (Direction == HIPFFT_C2R)
+      fftSafeCall(hipfftExecC2R(handle, data_complex_l, data_real_l));
+    else if constexpr (Direction == HIPFFT_D2Z)
+      fftSafeCall(hipfftExecD2Z(handle, data_real_l, data_complex_l));
+    else if constexpr (Direction == HIPFFT_Z2D)
+      fftSafeCall(hipfftExecZ2D(handle, data_complex_l, data_real_l));
+  }
+  void set_stream(cudaStream_t stream) {
```
Comment (samhatfield) [inline src/trans/gpu/algor/hicfft.hip.cpp:54]:
```suggestion
  void set_stream(hipStream_t stream) {
```

File: src/trans/gpu/CMakeLists.txt [hunk 2]
```diff
@@ -61,6 +61,7 @@ ecbuild_add_library(
                        $<${HAVE_CUTLASS}:USE_CUTLASS>
                        $<${HAVE_CUTLASS_3XTF32}:USE_CUTLASS_3XTF32>
                        $<${HAVE_GPU_GRAPHS_GEMM}:USE_GRAPHS_GEMM>
+		       $<${HAVE_GPU_GRAPHS_FFT}:USE_GRAPHS_FFT>
```
Comment (samhatfield) [inline src/trans/gpu/CMakeLists.txt:64]:
```suggestion
                       $<${HAVE_GPU_GRAPHS_FFT}:USE_GRAPHS_FFT>
```

Comment (samhatfield) [reply src/trans/gpu/CMakeLists.txt:64]: Sorry for being pedantic :) I just replaced the tab characters with spaces.

File: src/trans/gpu/CMakeLists.txt [hunk 3]
```diff
@@ -150,6 +151,7 @@ foreach( prec dp sp )
                                $<${HAVE_CUTLASS}:USE_CUTLASS>
                                $<${HAVE_CUTLASS_3XTF32}:USE_CUTLASS_3XTF32>
                                $<${HAVE_GPU_GRAPHS_GEMM}:USE_GRAPHS_GEMM>
+			       $<${HAVE_GPU_GRAPHS_FFT}:USE_GRAPHS_FFT>
```
Comment (samhatfield) [inline src/trans/gpu/CMakeLists.txt:154]:
```suggestion
                               $<${HAVE_GPU_GRAPHS_FFT}:USE_GRAPHS_FFT>
```

Ditto.

File: src/trans/gpu/CMakeLists.txt (hunk 2 above)
Comment (lukasm91) [reply src/trans/gpu/CMakeLists.txt:64]: It's good to be pedantic, and I am always for replacing characters with spaces :D

File: CMakeLists.txt [hunk 4]
```diff
@@ -140,6 +140,11 @@ ecbuild_add_option( FEATURE GPU_GRAPHS_GEMM
                     CONDITION HAVE_GPU
                     DESCRIPTION "Enable graph-based optimisation of Legendre transform GEMM kernel" )
 
+	    ecbuild_add_option( FEATURE GPU_GRAPHS_FFT
+                    DEFAULT ON
+                    CONDITION HAVE_GPU
+		    DESCRIPTION "Enable graph-based optimisation of Legendre transform FFT kernel" )
```
Comment (marsdeno) [inline CMakeLists.txt:146]: Trivial, but can you remove the "Legendre transform" here?

**Review by marsdeno — APPROVED** [review]
Looks great, thanks!

File: CMakeLists.txt (hunk 4 above)
Comment (lukasm91) [inline CMakeLists.txt:146]:
```suggestion
		    DESCRIPTION "Enable graph-based optimisation of FFT kernels" )
```

Comment (lukasm91) [reply CMakeLists.txt]: Makes sense, done.

**Pull request merged**